File: go-client-sdk/pkg/models/operations/getcompaniescompanyiddatafinancialsprofitandloss.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/v2/pkg/models/shared"
	"time"
)

type GetCompaniesCompanyIDDataFinancialsProfitAndLossPathParams struct {
	CompanyID string `pathParam:"style=simple,explode=false,name=companyId"`
}

type GetCompaniesCompanyIDDataFinancialsProfitAndLossQueryParams struct {
	PeriodLength     int        `queryParam:"style=form,explode=true,name=periodLength"`
	PeriodsToCompare int        `queryParam:"style=form,explode=true,name=periodsToCompare"`
	StartMonth       *time.Time `queryParam:"style=form,explode=true,name=startMonth"`
}

type GetCompaniesCompanyIDDataFinancialsProfitAndLossSecurity struct {
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
}

type GetCompaniesCompanyIDDataFinancialsProfitAndLossRequest struct {
	PathParams  GetCompaniesCompanyIDDataFinancialsProfitAndLossPathParams
	QueryParams GetCompaniesCompanyIDDataFinancialsProfitAndLossQueryParams
	Security    GetCompaniesCompanyIDDataFinancialsProfitAndLossSecurity
}

type GetCompaniesCompanyIDDataFinancialsProfitAndLossResponse struct {
	CodatPublicAPIModelsDataProfitAndLossResponse *shared.CodatPublicAPIModelsDataProfitAndLossResponse
	ContentType                                   string
	StatusCode                                    int
}

File: typescript-client-sdk/src/sdk/models/operations/getcompaniescompanyiddatafinancialsprofitandloss.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";


export class GetCompaniesCompanyIdDataFinancialsProfitAndLossPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=companyId" })
  companyId: string;
}

export class GetCompaniesCompanyIdDataFinancialsProfitAndLossQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=periodLength" })
  periodLength: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=periodsToCompare" })
  periodsToCompare: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=startMonth" })
  startMonth?: Date;
}

export class GetCompaniesCompanyIdDataFinancialsProfitAndLossSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeAPIKeyAuth;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;
}

export class GetCompaniesCompanyIdDataFinancialsProfitAndLossRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetCompaniesCompanyIdDataFinancialsProfitAndLossPathParams;

  @SpeakeasyMetadata()
  queryParams: GetCompaniesCompanyIdDataFinancialsProfitAndLossQueryParams;

  @SpeakeasyMetadata()
  security: GetCompaniesCompanyIdDataFinancialsProfitAndLossSecurity;
}

export class GetCompaniesCompanyIdDataFinancialsProfitAndLossResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  codatPublicApiModelsDataProfitAndLossResponse?: shared.CodatPublicApiModelsDataProfitAndLossResponse;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}